### ISSUE TITLE
Events are shown multiple times in the calendar.

### ISSUE THREAD

**Issue by rbhausoul** [humhub-contrib/external_calendar#56]
Over time, automatic synchronization leads to duplicate events in the calendar. So after a few days an event is listed 4x or more together.

I always solve this by doing a manual synchronization. Then each event is shown only once (as it should be). However, after a while the events double, due to the automatic synchronization.

The external calendar I sync is a Google Calendar.

Is this a bug or am I missing a setting to avoid events being shown twice in the calendar?

**Comment by molle100** [comment]
Must definitely a bug. Have the same issue since weeks when syncing an external calendar. Hourly sync meas 24x the same entry per day. Please fix this!

**Comment by marc-farre** [comment]
Duplicate of https://github.com/humhub-contrib/external_calendar/issues/51 ?

**Comment by marc-farre** [comment]
Original conversation: https://community.humhub.com/content/perma?id=269594

**Comment by rbhausoul** [comment]
Link is not working... "Oooops... It looks like you may have taken the wrong turn."

**Comment by samuk** [comment]
I also have this duplicate problem. Following

**Comment by kiwiradler** [comment]
Almost 6 months with the problem now - please help with that. It’s very annoying during daily work…

**Comment by marc-farre** [comment]
@kiwiradler could you provide detailed instructions to reproduce the issue?
If it's with a calendar you cannot share publicly but it's OK to share it with me and other Humhub developers, you can send it to me by private message: https://community.humhub.com/u/cuzy.app/
Thanks.

**Comment by kiwiradler** [comment]
Hi Marc,
I've just contacted you via Humhub Community. Thx for your help!

> @kiwiradler could you provide detailed instructions to reproduce the issue? If it's with a calendar you cannot share publicly but it's OK to share it with me and other Humhub developers, you can send it to me by private message: https://community.humhub.com/u/cuzy.app/ Thanks.

**Comment by marc-farre** [comment]
@luke- I now have a way to reproduce the bug.
@kiwiradler gave me the URL of an ICS file which I used to add an external calendar:
![image](https://github.com/humhub-contrib/external_calendar/assets/23310825/69cd36d2-d334-41e9-bdca-b117b00f1a31)

Each time I start an hourly cron job, all events are imported again. E.g. if I do 2 hourly cron jobs with:
`php yii cron/hourly; php yii queue/run -v; php yii cron/hourly; php yii queue/run -v;`
I have:
![image](https://github.com/humhub-contrib/external_calendar/assets/23310825/df372354-93cc-4f1b-84cb-eca61eb7c35d)

**Comment by kiwiradler** [comment]
Perfect! Thanks a lot for your work. 
One more thing: If you update both modules and copy the external calendar link into the module again (to remove the duplicated events), no more events are shown. Sync seems to work, but calendar is empty then.  I solved the problem with deleting both modules (calendar and external calendar) and set them up again from the scratch. Could be a problem if you have many spaces with different calendars - because deleting the modules will destroy all of your data. Your users won't very likely love that ;) For me no problem with only one calendar, but maybe you can change this in upcoming versions.
